lib/srv/desktop/windows_server.go

```diff
@@ -557,36 +557,40 @@ func (s *WindowsService) Serve(plainLis net.Listener) error {
 // It authenticates and authorizes the connection, and then begins
 // translating the TDP messages from the proxy into native RDP.
 func (s *WindowsService) handleConnection(proxyConn *tls.Conn) {
-	defer proxyConn.Close()
 	log := s.cfg.Log
+
 	tdpConn := tdp.NewConn(proxyConn)
```

Comment:
Suggestion:

```suggestion
	tdpConn := tdp.NewTrackingConn(proxyConn)
```

where NewTrackingConn is:

```go
type trackingConn interface {
	io.ReadWriter
	srv.TrackingConn
}

func NewTrackingConn(c trackingConn) *Conn { ... }
```

We could even make the Local and Remote fields private, or make separate conn types (as suggested in another comment).

Reply:
Not a huge fan of an exported constructor that accepts an unexported interface as an argument..

Reply:
Ohh I know what I'll do.

- I'll keep the signature the same.
- I'll unexport local/remote fields
- In the constructor I'll do a type assertion on the readwriter you provided, and if it has a `LocalAddr()` and `RemoteAddr()` method I will set the private fields.

This way, you get a tracking conn if you're wrapping a tracking conn. And if you're wrapping something like a bytes.Buffer for tests, you'll just get empty local/remote addrs.

Reply:
Sure, sounds alright. Friendly reminder to note this behavior in the comments.